### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by jmcarcell**
BEGINRELEASENOTES
- Fix compiler warnings with clang about `using namespace` in headers and `throw()` being deprecated
- Change to using `LCIO::lcio` instead of `${LCIO_LIBRARIES}`
- Fix a couple of CMake warnings due to incomplete file names

ENDRELEASENOTES

Some examples of the warnings:

``` c++
/LCCD/source/test/./CalibrationConstant.hh:15:17: warning: using namespace directive in global context in header [-Wheader-hygiene]
using namespace lcio ;
```

and 

``` c++
/LCCD/source/include/lccd_exceptions.h:59:42: warning: dynamic exception specifications are deprecated [-Wdeprecated-dynamic-exception-spec]
    virtual ~DataNotAvailableException() throw() { /*no_op*/; }
                                         ^~~~~~~
LCCD/source/include/lccd_exceptions.h:59:42: note: use 'noexcept' instead
    virtual ~DataNotAvailableException() throw() { /*no_op*/; }
                                         ^~~~~~~
                                         noexcept
```

### CODE REVIEW

File: source/include/lccd_exceptions.h [hunk 1]
```diff
@@ -84,7 +84,7 @@ namespace lccd{
   class InconsistencyException : public LCCDException{
 
   public: 
-    virtual ~InconsistencyException() throw() { /*no_op*/; } 
+    virtual ~InconsistencyException() noexcept { /*no_op*/; }
```
Comment (tmadlener) [inline source/include/lccd_exceptions.h]:
```suggestion
    virtual ~InconsistencyException() noexcept = default;
```

File: source/include/lccd_exceptions.h [hunk 2]
```diff
@@ -70,7 +70,7 @@ namespace lccd{
   class ReadOnlyException : public LCCDException{
 
   public: 
-    virtual ~ReadOnlyException() throw() { /*no_op*/; } 
+    virtual ~ReadOnlyException() noexcept { /*no_op*/; }
```
Comment (tmadlener) [inline source/include/lccd_exceptions.h]:
```suggestion
    virtual ~ReadOnlyException() noexcept = default;
```

File: source/include/lccd_exceptions.h [hunk 3]
```diff
@@ -56,7 +56,7 @@ namespace lccd{
   class DataNotAvailableException : public LCCDException{
 
   public: 
-    virtual ~DataNotAvailableException() throw() { /*no_op*/; } 
+    virtual ~DataNotAvailableException() noexcept { /*no_op*/; }
```
Comment (tmadlener) [inline source/include/lccd_exceptions.h]:
```suggestion
    virtual ~DataNotAvailableException() noexcept = default;
```

File: source/include/lccd_exceptions.h [hunk 4]
```diff
@@ -42,7 +42,7 @@ namespace lccd{
   protected:
     DatabaseException() {  /*no_op*/ ; } 
   public: 
-    virtual ~DatabaseException() throw() { /*no_op*/; } 
+    virtual ~DatabaseException() noexcept { /*no_op*/; }
```
Comment (tmadlener) [inline source/include/lccd_exceptions.h]:
```suggestion
    virtual ~DatabaseException() noexcept = default;
```

File: source/include/lccd_exceptions.h [hunk 5]
```diff
@@ -24,13 +24,13 @@ namespace lccd{
     LCCDException(){  /*no_op*/ ; } 
       
   public: 
-    virtual ~LCCDException() throw() { /*no_op*/; } 
+    virtual ~LCCDException() noexcept { /*no_op*/; }
```
Comment (tmadlener) [inline source/include/lccd_exceptions.h]:
```suggestion
    virtual ~LCCDException() noexcept = default;
```

File: source/include/lccd_exceptions.h [hunk 6]
```diff
@@ -98,7 +98,7 @@ namespace lccd{
   class MemberNotImplementedException : public LCCDException{
 
   public: 
-    virtual ~MemberNotImplementedException() throw() { /*no_op*/; } 
+    virtual ~MemberNotImplementedException() noexcept { /*no_op*/; }
```
Comment (tmadlener) [inline source/include/lccd_exceptions.h]:
```suggestion
    virtual ~MemberNotImplementedException() noexcept = default;
```

**Review by tmadlener — COMMENTED** [review]
I think `noexcept` destructors can still be marked `default`.

Also we could think about bumping the LCIO required version to have the targets.

File: source/include/lccd_exceptions.h (hunk 1 above)
Comment (jmcarcell) [reply source/include/lccd_exceptions.h]: All done

**Review by tmadlener — APPROVED** [review]

**Pull request merged**